File: dart/transaction_system/dart_application_1/bin/inventory_item.dart
// lib/models/inventory_item.dart

class InventoryItem {
  final String id;
  final String sku;
  final String name;
  final int quantity;
  final double unitCost;
  final DateTime lastUpdated;
  final String? description;

  const InventoryItem({
    required this.id,
    required this.sku,
    required this.name,
    required this.quantity,
    required this.unitCost,
    required this.lastUpdated,
    this.description,
  });

  InventoryItem copyWith({
    String? id,
    String? sku,
    String? name,
    int? quantity,
    double? unitCost,
    DateTime? lastUpdated,
    String? description,
  }) {
    return InventoryItem(
      id: id ?? this.id,
      sku: sku ?? this.sku,
      name: name ?? this.name,
      quantity: quantity ?? this.quantity,
      unitCost: unitCost ?? this.unitCost,
      lastUpdated: lastUpdated ?? this.lastUpdated,
      description: description ?? this.description,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'sku': sku,
        'name': name,
        'quantity': quantity,
        'unit_cost': unitCost,
        'last_updated': lastUpdated.toIso8601String(),
        'description': description,
      };

  factory InventoryItem.fromJson(Map<String, dynamic> json) => InventoryItem(
        id: json['id'] as String,
        sku: json['sku'] as String,
        name: json['name'] as String,
        quantity: json['quantity'] as int,
        unitCost: (json['unit_cost'] as num).toDouble(),
        lastUpdated: DateTime.parse(json['last_updated'] as String),
        description: json['description'] as String?,
      );

  @override
  String toString() =>
      'InventoryItem(id: $id, sku: $sku, name: $name, qty: $quantity, cost: $unitCost)';
}

File: python system/inventory_system/models/inventory_item.py
from datetime import datetime
from typing import Optional, Dict, Any


class InventoryItem:
    def __init__(
        self,
        id: str,
        sku: str,
        name: str,
        quantity: int,
        unit_cost: float,
        last_updated: datetime,
        description: Optional[str] = None,
    ):
        self.id = id
        self.sku = sku
        self.name = name
        self.quantity = quantity
        self.unit_cost = unit_cost
        self.last_updated = last_updated
        self.description = description

    def copy_with(
        self,
        id: Optional[str] = None,
        sku: Optional[str] = None,
        name: Optional[str] = None,
        quantity: Optional[int] = None,
        unit_cost: Optional[float] = None,
        last_updated: Optional[datetime] = None,
        description: Optional[str] = None,
    ) -> "InventoryItem":
        return InventoryItem(
            id=id if id is not None else self.id,
            sku=sku if sku is not None else self.sku,
            name=name if name is not None else self.name,
            quantity=quantity if quantity is not None else self.quantity,
            unit_cost=unit_cost if unit_cost is not None else self.unit_cost,
            last_updated=last_updated if last_updated is not None else self.last_updated,
            description=description if description is not None else self.description,
        )

    def to_dict(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "sku": self.sku,
            "name": self.name,
            "quantity": self.quantity,
            "unit_cost": self.unit_cost,
            "last_updated": self.last_updated.isoformat(),
            "description": self.description,
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> "InventoryItem":
        return cls(
            id=data["id"],
            sku=data["sku"],
            name=data["name"],
            quantity=data["quantity"],
            unit_cost=float(data["unit_cost"]),
            last_updated=datetime.fromisoformat(data["last_updated"]),
            description=data.get("description"),
        )

    def __str__(self) -> str:
        return f"InventoryItem(id={self.id}, sku={self.sku}, name={self.name}, qty={self.quantity}, cost={self.unit_cost})"